WordPress/Classes/ViewRelated/Blog/SharingViewController.m

```diff
@@ -95,7 +113,21 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
         case SharingPublicize: {
             Publicizer *publicizer = self.publicizeServices[indexPath.row];
             cell.textLabel.text = publicizer.label;
-            } break;
+
+            UIButton *button = [UIButton buttonWithType:UIButtonTypeSystem];
+            [button addTarget:self action:@selector(cellButtonTapped:) forControlEvents:UIControlEventTouchUpInside];
+            NSString *title = nil;
+            if ([self.connectingService.service isEqualToString:publicizer.service]) {
+                title = NSLocalizedString(@"Connecting…", @"Button title while a Publicize service is connecting");
+            } else if (publicizer.isConnected) {
+                title = NSLocalizedString(@"Disconnect", @"Button title to disconnect a Publicize service");
+            } else {
+                title = NSLocalizedString(@"Connect", @"Button title to connect a Publicize service");
+            }
+            [button setTitle:title forState:UIControlStateNormal];
+            [button sizeToFit];
+            cell.accessoryView = button;
```

Comment:
I know, definitely minimum impact, but this is gonna be alloc'ed each time the cell is dequeued (scroll, BG > FG perhaps). Maybe add a guard to do this once, or just switch to a nib?